utils/rag/rag_components.py

```diff
@@ -7,100 +9,298 @@
         self.example_judge = example_judge_prompt | self.llm | JsonOutputParser()
 
     def init_reform_chain(self) -> None:
+        """
+        Initializes the reformulation chain by loading the reformulation prompt and combining it with the language model and a string output parser.
 
-        reformulation_prompt = load_prompt(self.prompts_paths["reformulation_prompt.yaml"])
+        Args:
+            self (object): The current object instance.
+
+        Returns:
+            The self.reformulation_chain attribute.
+        """
+
+        reformulation_prompt = load_prompt(repo_dir + "/" + self.prompts_paths["reformulation_prompt"])
         self.reformulation_chain = reformulation_prompt | self.llm | StrOutputParser()
 
     def init_entity_chain(self) -> None:
+        """
+        Initializes the entity chain by loading the entity prompt and combining it with the language model and a JSON output parser.
+
+        Args:
+            self (object): The current object instance.
+
+        Returns:
+            The self.entity_chain attribute.
+        """
         
-        entity_prompt = load_prompt(self.prompts_paths["entity_prompt.yaml"])
+        entity_prompt = load_prompt(repo_dir + "/" + self.prompts_paths["entity_prompt"])
         self.entity_chain = entity_prompt | self.llm | JsonOutputParser()
 
     def init_subquery_chain(self) -> None:
+        """
+        Initializes the subquery chain by loading the subquery prompt and combining it with the language model and string output parser.
 
-        subquery_prompt = load_prompt(self.prompts_paths["subquery_prompt"])
+        Args:
+            self (object): The current object instance.
+
+        Returns:
+            The self.subquery_chain attribute.
+        """
+
+        subquery_prompt = load_prompt(repo_dir + "/" + self.prompts_paths["subquery_prompt"])
         self.subquery_chain = subquery_prompt | self.llm | StrOutputParser()
 
     def init_retrieval_grader(self) -> None:
+        """
+        Initializes the retrieval grader component.
+
+        This method loads the retrieval grader prompt from the repository and combines it with the language model and a JSON output parser.
+
+        Args:
+            self (object): The current object instance.
+
+        Returns:
+            The self.retrieval_grader attribute.
+        """
 
-        retrieval_grader_prompt = load_prompt(self.prompts_paths["retrieval_grader_prompt"])
+        retrieval_grader_prompt = load_prompt(repo_dir + "/" + self.prompts_paths["retrieval_grader_prompt"])
         self.retrieval_grader = retrieval_grader_prompt | self.llm | JsonOutputParser()
 
     def init_qa_chain(self) -> None:
+        """
+        Initializes the QA chain by loading the QA prompt and combining it with the large language model and a string output parser.
+
+        Args:
+            self (object): The current object instance.
 
-        qa_prompt = load_prompt(self.prompts_paths["qa_prompt"])
+        Returns:
```

Comment:
Not a return.

Reply:
Also below.

Reply:
True.  Would you like me to specify Return as None for whenever an attribute is set?